### ISSUE TITLE
doxygen error on plugin cpp.doxygen

### ISSUE THREAD

**Issue by aotto1968** [vim/vim#10604]
### Steps to reproduce

testfile name _test.tmpl_
```
/**
some text
**/

// vim:syntax=cpp.doxygen
```

create an error on open: `vim test.tmpl`

```
"test.tmpl" 7L, 47B
Error detected while processing modelines[7]..Syntax Autocommands for "*"..function <SNR>3_SynSet[25]..script /usr/share/vim/vim82/syntax/doxygen.vim[627]..modelines[7]..Syntax Autocommands for "*"..function <SNR>3_SynSet[25]..script /usr/share/vim/vim8
2/syntax/doxygen.vim:
line  502:
E12: Command not allowed from exrc/vimrc in current dir or tag search:     aug doxygengroup
```

### Expected behaviour

no error on startup!

If I follow the instructions and open `/usr/share/vim/vim82/syntax/doxygen.vim[627]`  I get…

```
    " This is still a proposal, but won't do any harm.
    aug doxygengroup
    au!
    au Syntax UserColor_reset nested call s:Doxygen_Hilights_Base()
    au Syntax UserColor_{on,reset,enable} nested call s:Doxygen_Hilights()
    aug END
```

Funny are the words: _but won't do any harm_

→ well if I comment out this block it works without error.

Error: The problem is that any-time my opensuse linux update the software the BUG arrive again.

### Version of Vim

VIM - Vi IMproved 8.2 (2019 Dec 12) Included patches: 1-5083 Compiled by 'http://www.opensuse.org/'

### Environment

1. Linux linux02 5.18.4-1-default #1 SMP PREEMPT_DYNAMIC Wed Jun 15 06:00:33 UTC 2022 (ed6345d) x86_64 x86_64 x86_64 GNU/Linux
2. konsole
3. screen-256color

**Comment by brammool** [comment]
Yeah, that makes sense.  I think a lot has changed since I wrote that plugin! Including a lot of that sandboxing.Michael On 22 June 2022 8:00 pm, Bram Moolenaar ***@***.***> wrote:
> ### Steps to reproduce
> 
> testfile name _test.tmpl_
> ```
> /**
> some text
> **/
> 
> // vim:syntax=cpp.doxygen
> ```
> 
> create an error on open: `vim test.tmpl`
> 
> ```
> "test.tmpl" 7L, 47B
> Error detected while processing modelines[7]..Syntax Autocommands for "*"..function <SNR>3_SynSet[25]..script /usr/share/vim/vim82/syntax/doxygen.vim[627]..modelines[7]..Syntax Autocommands for "*"..function <SNR>3_SynSet[25]..script /usr/share/vim/vim8
> 2/syntax/doxygen.vim:
> line  502:
> E12: Command not allowed from exrc/vimrc in current dir or tag search:     aug doxygengroup
> ```
> 
> 
> ### Expected behaviour
> 
> no error on startup!
> 
> If I follow the instructions and open `/usr/share/vim/vim82/syntax/doxygen.vim[627]`  I get…
> 
> ```
>     " This is still a proposal, but won't do any harm.
>     aug doxygengroup
>     au!
>     au Syntax UserColor_reset nested call s:Doxygen_Hilights_Base()
>     au Syntax UserColor_{on,reset,enable} nested call s:Doxygen_Hilights()
>     aug END
> ```
> 
> Funny are the words: _but won't do any harm_
> 
> → well if I comment out this block it works without error.
> 
> Error: The problem is that any-time my opensuse linux update the
> software the BUG arrive again.

Strange that it took so long for someone to run into this.

The problem is that from a modeline you cannot execute auto commands, to
avoid any security leaks.

Best seems to add a try/catch around these lines:

    try
      aug doxygengroup
        au!
        au Syntax UserColor_reset nested call s:Doxygen_Hilights_Base()
        au Syntax UserColor_{on,reset,enable} nested call s:Doxygen_Hilights()
      aug END
    catch /E12:/
    endtry

What do you think Michael?


-- 
Our job was to build a computer information system for the branch banks.  We
were the perfect people for the job: Dean had seen a computer once, and I had
heard Dean talk about it.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by aotto1968** [comment]
> Strange that it took so long for someone to run into this.

Well I use this quite often BUT run a old linux I updated recently. The old linux has no problem with this **or** there was no `vim` update for the _old_ linux and the _correction_ was good for a while.

AFTER upgrade the old linux I get updates again and the bug arrives again and start to annoy.

**Comment by dlejay** [comment]
@chrisbra I think the Issue has been addressed in vim/vim@8cc5b55

**Comment by chrisbra** [comment]
thanks. Closing as resolved then.
